File: src/main/java/org/pac4j/play/deadbolt2/Pac4jRoleHandler.java
package org.pac4j.play.deadbolt2;

import be.objectify.deadbolt.java.models.Permission;
import play.libs.concurrent.HttpExecutionContext;

import java.util.Collections;
import java.util.List;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public interface Pac4jRoleHandler {

   default  CompletionStage<List<? extends Permission>> getPermissionsForRole(String clients, String roleName, HttpExecutionContext httpExecutionContext) {
       return CompletableFuture.completedFuture(Collections.emptyList());
    }
}

File: src/main/scala/org/pac4j/play/scala/deadbolt2/Pac4jRoleHandler.scala
package org.pac4j.play.scala.deadbolt2

import scala.concurrent.{ExecutionContext, Future}

trait Pac4jRoleHandler {
  def getPermissionsForRole(clients: String, roleName: String)
    (implicit ec: ExecutionContext): Future[List[String]]
}

class SimpleRoleHandler extends Pac4jRoleHandler {
  override def getPermissionsForRole(clients: String, roleName: String)
    (implicit ec: ExecutionContext): Future[List[String]] = Future(Nil)
}
